mypy/checkmember.py

```diff
@@ -17,6 +17,8 @@
 from mypy.semanal import self_type
 from mypy import messages
 from mypy import subtypes
+if False:  # import for forward declaration only
```

Comment:
I don't think you need the "if False" -- all you need is to use the f.q.name in the forward references.